dataflux_pytorch/benchmark/checkpointing/multinode/train.py

```diff
@@ -142,14 +143,19 @@ def main(ckpt_dir_path: str, ckpt_restore_path: str = ""):
     )
     trainer.fit(model, dataloader)
     print(f"Saving checkpoint to {ckpt_dir_path} {num_save_calls} times.")
-    start = time.time()
+    save_checkpoint_times = []
```

Comment:
Also consider using [torch.cuda.Event()](https://pytorch.org/docs/stable/generated/torch.cuda.Event.html) to `record()` and measure `elapsed_time()`. 

See an example usage in section 2.4 of https://pytorch.org/tutorials/intermediate/FSDP_tutorial.html#how-to-use-fsdp

Whatever you decide here, I'll update my PR #168 to match.

Reply:
Our calls to save/load checkpoints here already enforce synchronization at the end in the underlying pytorch distributed logic, so I don't think I need to do anything special. Your case might be different if the nodes aren't already synchronized at that point.